### PR TITLE
Allow to use currently installed ansible-core for devel and stable subcommands

### PR DESCRIPTION
**Pull request by felixfontein**
CC @mattclay

### CODE REVIEW

File: src/antsibull_docs/cli/antsibull_docs.py [hunk 1]
```diff
@@ -277,6 +277,10 @@ def parse_args(program_name: str, args: List[str]) -> argparse.Namespace:
                                          ' release of Ansible')
     devel_parser.add_argument('--pieces-file', default=DEFAULT_PIECES_FILE,
                               help='File containing a list of collections to include')
+    devel_parser.add_argument('--use-current-ansible-core', action='store_true',
```
Comment (gotmax23) [inline src/antsibull_docs/cli/antsibull_docs.py:280]: IMO, something like `--use-installed-ansible-core` would be better. `current` seems confusing here.

Comment (felixfontein) [reply src/antsibull_docs/cli/antsibull_docs.py]: Renamed in ab51201a32d2cfa3bd911afeab8ee7a9b82aa26b.

File: src/antsibull_docs/cli/antsibull_docs.py [hunk 2]
```diff
@@ -277,6 +277,10 @@ def parse_args(program_name: str, args: List[str]) -> argparse.Namespace:
                                          ' release of Ansible')
     devel_parser.add_argument('--pieces-file', default=DEFAULT_PIECES_FILE,
                               help='File containing a list of collections to include')
+    devel_parser.add_argument('--use-installed-ansible-core', action='store_true',
+                              help='Assumes that ansible-core is already installed and can be'
+                              ' used by calling `ansible`, `ansible-doc`, and `ansible-galaxy`'
+                              ' from $PATH.')
```
Comment (gotmax23) [inline src/antsibull_docs/cli/antsibull_docs.py:283]:
```suggestion
                              help='Assumes that ansible-core is already installed and can be'
                              ' used by calling `ansible`, `ansible-doc`, and `ansible-galaxy`'
                              ' from $PATH.'
                              ' By default, antsibull-docs installs ansible-core into a temporary venv.')
```

File: src/antsibull_docs/cli/antsibull_docs.py [hunk 3]
```diff
@@ -289,6 +293,10 @@ def parse_args(program_name: str, args: List[str]) -> argparse.Namespace:
     stable_parser.add_argument('--deps-file', required=True,
                                help='File which contains the list of collections and'
                                ' versions which were included in this version of Ansible')
+    stable_parser.add_argument('--use-installed-ansible-core', action='store_true',
+                               help='Assumes that ansible-core is already installed and can be'
+                               ' used by calling `ansible`, `ansible-doc`, and `ansible-galaxy`'
+                               ' from $PATH.')
```
Comment (gotmax23) [inline src/antsibull_docs/cli/antsibull_docs.py:299]:
```suggestion
                              help='Assumes that ansible-core is already installed and can be'
                              ' used by calling `ansible`, `ansible-doc`, and `ansible-galaxy`'
                              ' from $PATH.'
                              ' By default, antsibull-docs installs ansible-core into a temporary venv.')
```

**Review by gotmax23 — APPROVED** [review]
One other small docs suggestion. Otherwise, this looks good to me. Thanks!

**Review by mattclay — CHANGES_REQUESTED** [review]
It seems this PR is still missing one necessary change. It doesn't look like it provides a way to preserve `PYTHONPATH`, which is being removed here:

https://github.com/ansible-community/antsibull-docs/blob/489da646f9784796a75e06426831e5c8f3f4b412/src/antsibull_docs/docs_parsing/__init__.py#L34

This prevents antsibull-docs from recognizing the ansible-core that is injected into the virtual environment.

**Review by mattclay — APPROVED** [review]

**Comment by felixfontein** [comment]
@gotmax23 @mattclay thanks for reviewing this!

**Comment by mattclay** [comment]
@felixfontein Do you have an idea of when you'll be releasing a version with this change?

**Comment by felixfontein** [comment]
@mattclay first antsibull-core 2.0.0 needs to be released before antsibull-docs 2.0.0 can be released (which will contain this change). We're in the middle of preparing both; I'm right now not sure how long this will take. I hope/think/assume both will be done in a couple of weeks, but it depends a lot on when/how much @gotmax23 and me have time to work on it...

**Comment by gotmax23** [comment]
Is this worth backporting?

**Comment by felixfontein** [comment]
It's a feature, so no backport. `stable-1` is reserved for bugfixes (so we can update antsibull-docs in ansible-core 2.15).

**Pull request merged**